### ISSUE TITLE
Error with Pytest 6.0.0

### ISSUE THREAD

**Issue by brad-natelborg** [pytest-dev/pytest-html#321]
With the latest version of Pytest coming out on 7/29/2020, I now get the following error in pytest-html:

```
INTERNALERROR> Traceback (most recent call last):
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/main.py", line 240, in wrap_session
INTERNALERROR>     session.exitstatus = doit(config, session) or 0
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/main.py", line 296, in _main
INTERNALERROR>     config.hook.pytest_runtestloop(session=session)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/manager.py", line 87, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/main.py", line 321, in pytest_runtestloop
INTERNALERROR>     item.config.hook.pytest_runtest_protocol(item=item, nextitem=nextitem)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/manager.py", line 87, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/runner.py", line 100, in pytest_runtest_protocol
INTERNALERROR>     runtestprotocol(item, nextitem=nextitem)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/runner.py", line 111, in runtestprotocol
INTERNALERROR>     rep = call_and_report(item, "setup", log)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/runner.py", line 211, in call_and_report
INTERNALERROR>     hook.pytest_runtest_logreport(report=report)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/hooks.py", line 286, in __call__
INTERNALERROR>     return self._hookexec(self, self.get_hookimpls(), kwargs)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/manager.py", line 93, in _hookexec
INTERNALERROR>     return self._inner_hookexec(hook, methods, kwargs)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/manager.py", line 87, in <lambda>
INTERNALERROR>     firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/callers.py", line 208, in _multicall
INTERNALERROR>     return outcome.get_result()
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/callers.py", line 80, in get_result
INTERNALERROR>     raise ex[1].with_traceback(ex[2])
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pluggy/callers.py", line 187, in _multicall
INTERNALERROR>     res = hook_impl.function(*args)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pytest_html/plugin.py", line 559, in pytest_runtest_logreport
INTERNALERROR>     self.append_skipped(report)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pytest_html/plugin.py", line 336, in append_skipped
INTERNALERROR>     self._appendrow("Skipped", report)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pytest_html/plugin.py", line 296, in _appendrow
INTERNALERROR>     result = self.TestResult(outcome, report, self.logfile, self.config)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pytest_html/plugin.py", line 137, in __init__
INTERNALERROR>     self.append_log_html(report, self.additional_html)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/pytest_html/plugin.py", line 268, in append_log_html
INTERNALERROR>     for line in report.longreprtext.splitlines():
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/reports.py", line 105, in longreprtext
INTERNALERROR>     self.toterminal(tw)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/reports.py", line 85, in toterminal
INTERNALERROR>     out.line(longrepr)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/_io/terminalwriter.py", line 168, in line
INTERNALERROR>     self.write(s, **markup)
INTERNALERROR>   File "/usr/local/lib/python3.5/dist-packages/_pytest/_io/terminalwriter.py", line 144, in write
INTERNALERROR>     current_line = msg.rsplit("\n", 1)[-1]
INTERNALERROR> AttributeError: 'tuple' object has no attribute 'rsplit'
```

**Comment by jorge-dev** [comment]
Having same Issue. In my case the error only happens when a test is skipped by using `@pytest.mark.skipif`.
Doing some digging I found the error is most likely originated from TestResult class inside `pytest_html/plugin.py`.
Commenting line 380 in same file allows the test to run and suppresses the error. However, the skipped tests will not be displayed in html report table.

![image](https://user-images.githubusercontent.com/20519640/88855818-aa4f6280-d1b0-11ea-9d07-4bae339171ae.png)
![image](https://user-images.githubusercontent.com/20519640/88855851-bc310580-d1b0-11ea-8c14-81d33f978c1e.png)

**Comment by BeyondEvil** [comment]
Revert to latest 5 version, fix is coming in 6.0.1

**Comment by ssbarnea** [comment]
Master already avoids 6.0.0 version and it does not make sense to force a release. We can close the bug as upstream one, one with simple workarounds.

**Comment by brad-natelborg** [comment]
@ssbarnea When you say simple workarounds, are you just referring to using an older pytest version?

**Comment by ssbarnea** [comment]
Take a look at https://github.com/pytest-dev/pytest-html/pull/319/files -- if I remember well, master branch of pytest already had it fixed, which means wait for 6.0.1 version. There is nothing else than skipping 6.0.0 from the pytest-html point of view.

Or just wait and likely it will sort-out by magic, when 6.0.1 is released ;)

To prevent future surprises I already proposed https://github.com/pytest-dev/pytest-html/pull/320

**Comment by brad-natelborg** [comment]
Got it, thanks!
